lib/active_merchant/billing/gateways/payment_highway.rb

```diff
@@ -0,0 +1,172 @@
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class PaymentHighwayGateway < Gateway
+      self.money_format = :cents
+      self.test_url = 'https://v1-hub-staging.sph-test-solinor.com/'
+      self.live_url = 'https://v1.api.paymenthighway.io'
+
+      self.supported_countries = ['FI']
+      self.default_currency = 'EUR'
+      self.supported_cardtypes = [:visa, :master, :american_express, :discover]
+
+      self.homepage_url = 'https://www.paymenthighway.fi/'
+      self.display_name = 'PaymentHighway'
+
+      STANDARD_ERROR_CODE_MAPPING = {}
+
+      RESPONSE_CODE_MAPPING = {
+        100 => 'Request successful.',
+        200 => 'Authorization failed',
+        901 => 'Invalid input. Detailed information is in the message field.'
+      }
+
+      def initialize(options={})
+        requires!(options, :sph_account, :sph_merchant, :account_key, :account_secret)
+        @sph_account = options[:sph_account]
+        @sph_merchant = options[:sph_merchant]
+        @account_key = options[:account_key]
+        @account_secret = options[:account_secret]
+        super
+      end
+
+      def purchase(money, card, options={})
+        transaction_id = initTransaction
+
+        payload = {
+          'amount' => money,
+          'currency' => default_currency,
+          'card' => card_to_json(card)
+        }
+
+        payload = add_ip(payload, options)
+        payload = add_order_id(payload, options)
+
+        commit("/transaction/#{transaction_id}/debit", payload.to_json, transaction_id)
+      end
+
+      def order_status(order_id)
```

Comment:
We don't encourage exposing methods like this in the public part of the class API.